### PR TITLE
Feature: Added Github codespaces configurations #1541

### PR DESCRIPTION
**Pull request by Shrest4647**
Related issue: Issue #1541

Change Logs:
- Fixed the volume mount issues in the dev container configurations
- Updated the postAttachCommand to start the rails server.
- Updated the docker file permissions for installing packages

### CODE REVIEW

**Comment by apoorv1316** [comment]
@akhilgkrishnan Please review

File: .devcontainer/script.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+bin/rails db:prepare
```
Comment (akhilgkrishnan) [inline .devcontainer/script.sh:1]: Seems like we are not using this file. Or we can use this file by moving all the `postAttachCommand` values here and run this script inside the postAttachCommand.

**Review by akhilgkrishnan — COMMENTED** [review]
@Shrest4647, This is really great 🎉 . Thanks for working on this :)

I have a small suggestion, and in addition, we should modify the `development.rb` configuration to allow requests from the CodeSpaces URL. To do this, please add the following code block inside the `development.rb` file:

```ruby
if ENV["CODESPACES"] == "true"
  codespace_domain = ENV['GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN']
  codespace_host = "#{ENV['CODESPACE_NAME']}-3000.#{codespace_domain}"
  config.hosts << codespace_host
end
```

**Comment by Shrest4647** [comment]
Hi @akhilgkrishnan . I have updated the configuration for development.rb. Also, I have decided not to move the `postAttachCommand` to script.sh as this requires changing file permissions to execute the script and this also makes the running commands opaque to the user.

**Comment by akhilgkrishnan** [comment]
> Hi @akhilgkrishnan . I have updated the configuration for development.rb. Also, I have decided not to move the `postAttachCommand` to script.sh as this requires changing file permissions to execute the script and this also makes the running commands opaque to the user.

Can you fix the rubocop issue?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
```
Comment (akhilgkrishnan) [inline .devcontainer/devcontainer.json:1]: Can you fix the the indentation?

**Review by akhilgkrishnan — COMMENTED** [review]
```
  
ActionController::InvalidAuthenticityToken (HTTP Origin header (http://localhost:3000) didn't match request.base_url (https://fluffy-space-zebra-v79wj99x7vw3wg6x-3000.app.github.dev)):
  
actionpack (7.0.8) lib/action_controller/metal/request_forgery_protection.rb:253:in `handle_unverified_request'
actionpack (7.0.8) lib/action_controller/metal/request_forgery_protection.rb:286:in `handle_unverified_request'
devise (4.8.1) lib/devise/controllers/helpers.rb:255:in `handle_unverified_request'
actionpack (7.0.8) lib/action_controller/metal/request_forgery_protection.rb:275:in `verify_authenticity_token'
activesupport (7.0.8) lib/active_support/callbacks.rb:400:in `block in make_lambda'
activesupport (7.0.8) lib/active_support/callbacks.rb:199:in `block (2 levels) in halting'
actionpack (7.0.8) lib/abstract_controller/callbacks.rb:34:in `block (2 levels) in <module:Callbacks>'
activesupport (7.0.8) lib/active_support/callbacks.rb:200:in `block in halting'
activesupport (7.0.8) lib/active_support/callbacks.rb:595:in `block in invoke_before'
activesupport (7.0.8) lib/active_support/callbacks.rb:595:in `each'
activesupport (7.0.8) lib/active_support/callbacks.rb:595:in `invoke_before'
activesupport (7.0.8) lib/active_support/callbacks.rb:116:in `block in run_callbacks'
activesupport (7.0.8) lib/active_support/callbacks.rb:138:in `run_callbacks'
actionpack (7.0.8) lib/abstract_controller/callbacks.rb:233:in `process_action'
```

I can see this error on the logs, so avoiding this we might need to add below code in developement.rb

```ruby
  config.action_dispatch.default_headers = {
    'X-Frame-Options' => "ALLOW-FROM #{codespace_domain}"
  }
 ```
 
 Also we want to set the `APP_BASE_URL` env with codespace app url

**Comment by Animesh-Ghosh** [comment]
Good going! I was also kind of trying it out, but what I ended up doing was pointing the `dockerComposeFile` to the `docker-compose.yml` file in the project root. Codespace pre-build was successfull but I couldn't get a similar development experience as one would have locally.

[Here](https://github.com/Animesh-Ghosh/miru-web/commits/setup-gh-cs-devcontainer/) are the commits for reference.

Anyway, thanks for this! Will make contributions just one click away! 🚀

**Comment by Animesh-Ghosh** [comment]
@akhilgkrishnan apparently the error shown [here](https://github.com/saeloun/miru-web/pull/1557#pullrequestreview-1688150410) still exists when you try to submit a form/to a JSON end-point.
I created a Codespace from the GitHub UI from the `develop` branch and the server ran and I could open the homepage. However, when I tried to login, I faced the same error, along with a 422 Unprocessable Entity error.

Let me know if a separate issue should be created.

**Pull request merged**